[BLOG] Add Making pygments accessible

isabela-pf opened the pull request:
<!--Hi there! If you aren't submitting a blog post, you can delete this message.
Thank you for submitting a blog post! We want to make sure content produced for Quansight Labs is accessible, so please address the the following guidelines in your post.-->

## Text styling

- [x] The blog is written with [plain language](https://www.plainlanguage.gov/guidelines/) (where relevant).
- [x] If there are headers, they use the proper header tags in order to do so (with only one level-one header).
- [x] All links describe where they link to (for example, check the [Quansight labs website](https://labs.quansight.org/)).
- [x] Any kind of styling that the author uses (for example, bold for emphasis) is consistent throughout the blog.

## Non-text contents

- [x] All content is represented as text (for example, images need alt text and videos need captions or descriptive transcripts).
- [x] If there are emojis, there are not more than three in a row.
- [x] Don't use [flashing gifs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).
- [x] If it were to be read as plain text, the blog still makes sense and no information is missing.

---

This blog was requested by @trallard and written by @steff456 and myself. It announces Stephannie's work to make Pygments versions of existing accessible syntax highlighting themes and invites readers to try out the package. 

This is my first time adding a blog post to the new website, so please let me know if I made a mistake so I can learn for the future. Thanks in advance for any feedback! 🌻

asmeurer commented on apps/labs/posts/lets-make-pygments-accessible.md:
```diff
@@ -0,0 +1,70 @@
+
+The best accessibility experiences are often ones with strong defaults that welcome customization. In `accessible-pygments`, we are still focused on the strong defaults portion of that statement. Because we do not support customization, our goal is to [provide a growing list of curated accessible themes](https://github.com/Quansight-Labs/accessible-pygments/issues/2) for people to choose from. Ha ...
+
+As a note: general accessibility advice is to not rely exclusively on color to convey any information. Since syntax highlighting is all color information the authors have yet to find if there is an alternate approach.
```
But syntax highlighting doesn't convey any information that isn't already in the text itself. All it does it make the code easier to read by distinguishing the different types of code tokens.

isabela-pf replied:
This is true! No information is lost by not being able to make use of syntax highlighting for any reason, but the kind of support that it can provide for someone reading code is lost as far as I'm aware. I probably didn't make that distinction clear enough.

Do you want me to remove the note in line 46?

trallard replied:
Agree - no information is lost - _what does get affected though_ is more aligned with readability/scannability (if that is a word). 
Though - some IDE's and syntax highlighters _do_ allow the use of _italics_ to complement the "conveying information with colour"

asmeurer commented:
> - [x] Don't use [flashing gifs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).

This doesn't seem to be the case. The gifs here flash between the light and dark themes. Why not just show all the themes as multiple images?

steff456 commented:
> > - [x] Don't use [flashing gifs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).
> 
> 
> 
> This doesn't seem to be the case. The gifs here flash between the light and dark themes. Why not just show all the themes as multiple images? 

We could do any of them, I'm not sure which one will look better 😅

trallard reviewed (approved):
Some minor changes needed otherwise this looks good to me

trallard commented on apps/labs/posts/lets-make-pygments-accessible.md:
```diff
@@ -0,0 +1,70 @@
+description: 'accessible-pygments hosts curated WCAG-compliant themes for all your syntax highlighting needs.'
+category: [Access-centered]
+featuredImage:
+src: /posts/lets-make-pygments-accessible/featured_themes.gif
```
Per @isabela-pf comment elsewhere, this needs updating for the static version

isabela-pf commented:
Hi again! I think I've addressed all the feedback listed above and moved the post date out to November 7. Please let me know if there's anything else you'd like me to address. Thank you!

asmeurer commented:
Even ignoring the flashing concern, I prefer the style now used here because it makes it easier to compare the themes side-by-side.

asmeurer commented:
I have a few high-level comments relating to the post. I know this is somewhat close to the posting deadline, so feel free to not address these if it's too much work (although I am curious about the answers for my own edification regardless).

- The post doesn't seem to mention how the themes were created. Were they constructed from existing themes or created from scratch? If they were based on pre-existing themes were the color changes made by hand or using an automated tool?

- Are there any plans to upstream these themes to pygments, or is the plan to only keep them in this separate package for now?

Also, it looks like the vercel build failed because of some YAML issue in the post metadata.

trallard commented on apps/labs/posts/lets-make-pygments-accessible.md:
```diff
@@ -0,0 +1,69 @@
+description: 'accessible-pygments hosts curated WCAG-compliant themes for all your syntax highlighting needs.'
+category: [Access-centered]
+featuredImage:
+src: /posts/lets-make-pygments-accessible/featured_themes.png
```
```suggestion
  src: /posts/lets-make-pygments-accessible/featured_themes.png
```

trallard commented on apps/labs/posts/lets-make-pygments-accessible.md:
```diff
@@ -0,0 +1,69 @@
+alt: 'An animation of different accessible syntax highlighting themes being applied to the same example code.'
+hero:
+imageSrc: /posts/lets-make-pygments-accessible/hero_themes.png
+imageAlt: 'An animation of different accessible syntax highlighting themes being applied to the same example code.'
```
```suggestion
  imageAlt: 'An animation of different accessible syntax highlighting themes being applied to the same example code.'
```

trallard commented on apps/labs/posts/lets-make-pygments-accessible.md:
```diff
@@ -0,0 +1,69 @@
+---
+title: 'Let's make Pygments accessible!'
+author: stephannie-jimenez
+published: November 7, 2022
```
```suggestion
published: November 15, 2022
```

Moving a week later as we are deploying on Tuesdays and have a new post for this week

noatamir commented on apps/labs/posts/lets-make-pygments-accessible.md:
```diff
@@ -0,0 +1,70 @@
+---
+title: '"Let's make Pygments accessible!"'
```
```suggestion
title: "Let's make Pygments accessible!"
```

noatamir replied:
I think the double quotes might be the issue here.

trallard replied:
It was with single quotes before so will need inspecting closer

noatamir replied:
Sorry, poorly phrased! I meant double as in double within single.

noatamir replied:
The author slug is correct. I checked that before commenting. This is what made me investigate the row above to begin with.

trallard replied:
Yup we have tried few combos of the quotes. 

@isabela-pf and @steff456  since it has been too much faff debugging this what about changing the title to "Making pygments accessible" 🙏🏽 ? At least to avoid delaying this once more

trallard commented on apps/labs/posts/lets-make-pygments-accessible.md:
```diff
@@ -0,0 +1,70 @@
+category: [Access-centered]
+featuredImage:
+  src: /posts/lets-make-pygments-accessible/featured_themes.png
+  alt: 'An animation of different accessible syntax highlighting themes being applied to the same example code.'
```
```suggestion
  alt: 'A display of different accessible syntax highlighting themes being applied to the same example code.'
```

since it is not an animation anymore

trallard commented on apps/labs/posts/lets-make-pygments-accessible.md:
```diff
@@ -0,0 +1,70 @@
+  alt: 'An animation of different accessible syntax highlighting themes being applied to the same example code.'
+hero:
+  imageSrc: /posts/lets-make-pygments-accessible/hero_themes.png
+  imageAlt: 'An animation of different accessible syntax highlighting themes being applied to the same example code.'
```
```suggestion
  alt: 'A display of different accessible syntax highlighting themes being applied to the same example code.'
```

since it is not an animation anymore

trallard commented on apps/labs/posts/lets-make-pygments-accessible.md:
```diff
@@ -0,0 +1,70 @@
+
+Creating themes is HARD enough, and checking whether a theme has proper contrast may be unfamiliar to people less aware of accessibility needs and standards. Additionally, there is no “correct” answer, so we need multiple theming options.
+
+[Pygments](https://pygments.org/) is a very popular package for syntax highlighting and is the default backend for a lot of packages. Within the Jupyter ecosystem, a community [we have worked to improve a breadth of accessibility issues](https://jupyter-a11y.netlify.app/) in, also uses Pygments as its syntax highlighting backend for projects like [Jupyter Book](https://jupyterbook.org/en/stable/i ...
```
```suggestion
[Pygments](https://pygments.org/) is a very popular package for syntax highlighting and is the default backend for many packages. Within the Jupyter ecosystem, a community [we have worked to improve a breadth of accessibility issues](https://jupyter-a11y.netlify.app/) in, also uses Pygments as its syntax highlighting backend for projects like [Jupyter Book](https://jupyterbook.org/en/stable/intro.html). This means that in interacting with code cells and blocks, you are interacting with potential color issues inherited throughout the ecosystem. Knowing that Jupyter projects are far from the only ones that could benefit from accessibility-considerate syntax highlighting, we created our package named `accessible-pygments` to host themes that already had accessible and WCAG-compliant colors. To date, this includes 16 brand new themes based on [Eric Bailey’s a11y-syntax-highlighting themes](https://github.com/ericwbailey/a11y-syntax-highlighting), [GitHub VSCode themes](https://github.com/primer/github-vscode-theme), [Gotthard VSCode theme](https://github.com/janbiasi/vscode-gotthard-theme/), [blinds theme](https://github.com/orbulant/blinds-theme), [greative](https://github.com/SumanKhdka/Greative-VSCode-Theme) and [Tania Allard’s Pitaya Smoothie theme](https://github.com/trallard/pitaya_smoothie).
```

trallard commented:
This is finally building and the content has been reviewed multiple times. Merging now 🚀
